src/test/unit/mixer_tricopter_unittest.cc

```diff
@@ -21,55 +21,162 @@
 #include <limits.h>
 
 extern "C" {
-  #include "platform.h"
-  #include "flight/mixer.h"
-  #include "flight/mixer_tricopter.h"
-  #include "io/beeper.h"
-  #include "io/rc_controls.h"
-  #include "config/runtime_config.h"
-  #include "debug.h"
-  #include "drivers/sensor.h"
-  #include "drivers/accgyro.h"
-  #include "common/axis.h"
-  #include "common/filter.h"
+#include "debug.h"
+
+#include "platform.h"
+
+#include "config/runtime_config.h"
+
+#include "common/axis.h"
+#include "common/filter.h"
+
+#include "drivers/sensor.h"
+#include "drivers/accgyro.h"
+
+#include "flight/mixer.h"
+#include "flight/mixer_tricopter.h"
+
+#include "io/beeper.h"
+#include "io/rc_controls.h"
+
+servoParam_t servoConf;
+mixerConfig_t mixerConfig;
+tailTune_t tailTune;
+int16_t servo[MAX_SUPPORTED_SERVOS];
 }
 
 #include "unittest_macros.h"
 #include "gtest/gtest.h"
 
+class BasicTest: public ::testing::Test {
+protected:
+    virtual void SetUp() {
+        memset(&servoConf, 0, sizeof(servoConf));
+        servoConf.min = DEFAULT_SERVO_MIN;
+        servoConf.max = DEFAULT_SERVO_MAX;
+        servoConf.middle = DEFAULT_SERVO_MIDDLE;
+        servoConf.rate = 100;
+        servoConf.forwardFromChannel = CHANNEL_FORWARDING_DISABLED;
+
+        // give all servos a default command
+        for (uint8_t i = 0; i < MAX_SUPPORTED_SERVOS; i++) {
+            servo[i] = DEFAULT_SERVO_MIDDLE;
+        }
+
+        triInitMixer(&servoConf, &servo[5], &mixerConfig);
+        tailTune.mode = TT_MODE_THRUST_TORQUE;
+        tailTune.tt.state = TT_WAIT_FOR_DISARM;
+        tailTune.tt.servoAvgAngle.numOf = 300;
+    }
+};
 
-TEST(ATest, test) {
-    ASSERT_TRUE(false);
+TEST_F(BasicTest, TestThrustFactorCalculation139) {
+    // given
+    // expecting tri_tail_motor_thrustfactor = 1 / tan(angle)
+    tailTune.tt.servoAvgAngle.sum = 12345;
```

Comment:
I understand, I don't want to be a nitpicker here, I'm so grateful that you wrote unit tests. Something that I should have done.

Just thinking that we could comment the formula in this first test case so the next one who reads through understands where did the magic number come from.